diff --git a/week3/lab1b/#Makefile# b/week3/lab1b/#Makefile#
index f343039..34bc5d3 100644
--- a/week3/lab1b/#Makefile#
+++ b/week3/lab1b/#Makefile#
@@ -5,13 +5,18 @@
 CC=gcc
 CFLAGS = -g
 
-all:
-	gcc lab1b-client.c -std=c99 -o lab1b-client
-	gcc server.c -std=c99 -o server
+all:	createserver createclient
+
 
 
+all:
+	gcc lab1b-client.c -g -lmcrypt -std=c99 -o lab1b-client
+	gcc lab1b-server.c -g -lmcrypt -std=c99 -o lab1b-server
+	gcc client.c -lmcrypt -std=c99 -o client
+	gcc server.c -lmcrypt -std=c99 -o server
+
 clean:
 	@rm -f lab1a *.txt *.tar.gz
 
 dist: 
-	@tar -czvf lab1a-204577214.tar.gz Makefile lab1a.c README
+	@tar -czvf lab1a-204577214.tar.gz Makefile lab1b-server.c lab1b-client.c README
diff --git a/week3/lab1b/lab1b-client b/week3/lab1b/lab1b-client
index f7322d4..fbcef34 100755
Binary files a/week3/lab1b/lab1b-client and b/week3/lab1b/lab1b-client differ
diff --git a/week3/lab1b/lab1b-client.c b/week3/lab1b/lab1b-client.c
index 57b30ec..ab6ce98 100644
--- a/week3/lab1b/lab1b-client.c
+++ b/week3/lab1b/lab1b-client.c
@@ -23,7 +23,9 @@ struct termios saved_attributes;
 struct pollfd fds[2];
 char buffer[2048];
 int logfd;
-//MCRYPT td;
+char* encopt = NULL;
+char key [1024];
+int keysize = 0;
 
 void sysFailed(char* sysCall, int exitNum)
 {
@@ -68,18 +70,13 @@ MCRYPT encryptInit(void)
 {
   MCRYPT td;
   int i;
-  char *key;
-  char password[20];
+
   char *IV;
-  int keysize=19; /* 128 bits */
 
-  key=calloc(1, keysize);
-  strcpy(password, "A_large_key");
 
   /* Generate the key using the password */
   /*  mhash_keygen( KEYGEN_MCRYPT, MHASH_MD5, key, keysize, NULL, 0, password, strlen(password));
    */
-  memmove( key, password, strlen(password));
 
   td = mcrypt_module_open("twofish", NULL, "cfb", NULL);
   if (td==MCRYPT_FAILED) {
@@ -119,7 +116,6 @@ int main(int argc, char *argv[]) {
   int encFlag = 0;
   int logFlag = 0;
   char* portopt = NULL;
-  char* encopt = NULL;
   char* logopt = NULL;
 
   static struct option long_options[] = {
@@ -164,6 +160,7 @@ int main(int argc, char *argv[]) {
   MCRYPT dtd;
   if (encFlag)
     {
+      int ecfd = open(encopt, O_RDONLY);
       etd = encryptInit();
       dtd = encryptInit();
     }
diff --git a/week3/lab1b/lab1b-server b/week3/lab1b/lab1b-server
index 3bd33a3..1a445f2 100755
Binary files a/week3/lab1b/lab1b-server and b/week3/lab1b/lab1b-server differ
diff --git a/week3/lab1b/lab1b-server.c b/week3/lab1b/lab1b-server.c
index 285d495..af099e9 100644
--- a/week3/lab1b/lab1b-server.c
+++ b/week3/lab1b/lab1b-server.c
@@ -25,7 +25,9 @@ int status;
 int newsockfd;
 char buffer[2048];
 struct pollfd fds[2];
-
+char* encopt = NULL;
+char key [1024];
+int keysize = 0;
 
 
 void sysFailed(char* sysCall, int exitNum)
@@ -61,18 +63,13 @@ MCRYPT encryptInit(void)
 {
   MCRYPT td;
   int i;
-  char *key;
-  char password[20];
-  char *IV;
-  int keysize=19; /* 128 bits */
 
-  key=calloc(1, keysize);
-  strcpy(password, "A_large_key");
+  char *IV;
+  //  int keysize=19; /* 128 bits */
 
   /* Generate the key using the password */
   /*  mhash_keygen( KEYGEN_MCRYPT, MHASH_MD5, key, keysize, NULL, 0, password, strlen(password));
    */
-  memmove( key, password, strlen(password));
 
   td = mcrypt_module_open("twofish", NULL, "cfb", NULL);
   if (td==MCRYPT_FAILED) {
@@ -109,7 +106,7 @@ int main( int argc, char *argv[] ) {
     int portFlag = 0;
     int encFlag = 0;
     char* portopt = NULL;
-    char* encopt = NULL;
+    //    char* encopt = NULL;
 
     static struct option long_options[] = {
       {"port", required_argument, 0, 'p'},
@@ -139,6 +136,8 @@ int main( int argc, char *argv[] ) {
     MCRYPT dtd;
     if (encFlag)
       {	
+	int encfd = open(encopt, O_RDONLY);
+	keysize = read(encfd, key, 1024);
 	etd = encryptInit();		
 	dtd = encryptInit();
       }
